### PR TITLE
Bids 3175/data access user creation

### PR DESCRIPTION
**Pull request by Eisei24**

### CODE REVIEW

File: backend/pkg/api/data_access/user.go [hunk 1]
```diff
@@ -96,13 +143,27 @@ func (d *DataAccessService) UpdatePasswordResetTime(ctx context.Context, userId
...
+	var result string
+
+	err := d.userReader.GetContext(ctx, &result, `
+    	SELECT
+			email_confirmation_hash
+		FROM users
+		WHERE id = $1`, userId)
+
+	return result, err
 }
```
Comment (remoterami) [inline backend/pkg/api/data_access/user.go:155]: I'm honestly not even sure why this method exists, I don't think it's needed. My bad, would you remove it? :)

**Review by remoterami — CHANGES_REQUESTED** [review]
ptal

File: backend/pkg/api/data_access/user.go [hunk 2]
```diff
@@ -37,43 +37,84 @@ type UserRepository interface {
...
-	// TODO @DATA-ACCESS
 	// (password is already hashed)
-	return d.dummy.UpdateUserPassword(ctx, userId, password)
+
+	_, err := d.userWriter.ExecContext(ctx, `
+		UPDATE users 
+		SET 
+			password = $1
+		WHERE id = $2
+	`, password, userId)
```
[the remark this answers is not in the document]
Comment (Eisei24) [reply backend/pkg/api/data_access/user.go:98]: Right, makes sense to have it like for the email.

File: backend/pkg/api/data_access/user.go (hunk 1 above)
Comment (Eisei24) [reply backend/pkg/api/data_access/user.go:155]: Sure, removed.

**Review by remoterami — APPROVED** [review]
lgtm

**Pull request merged**